### PR TITLE
Fix Provider Crash Due to Deleted Service in UI

### PR DESCRIPTION
**Pull request by masha1997**

### CODE REVIEW

File: provider/resource_kpi_base_search.go [hunk 1]
```diff
@@ -571,6 +574,7 @@ func (r *resourceKpiBaseSearch) Read(ctx context.Context, req resource.ReadReque
 
 func (r *resourceKpiBaseSearch) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
 	var plan KpiBaseSearchState
+	//var base *models.Base
```
Comment (benoittoulme) [inline provider/resource_kpi_base_search.go:577]: remove commented line?

File: provider/resource_kpi_threshold_template.go [hunk 2]
```diff
@@ -466,7 +466,11 @@ func (r *resourceKpiThresholdTemplate) Read(ctx context.Context, req resource.Re
 		return
 	}
 	if b == nil {
-		resp.Diagnostics.AddError("KPI threshold template is not found", fmt.Sprintf("KPI threshold template %s (%s) is not found", state.ID.ValueString(), state.Title.ValueString()))
```
Comment (benoittoulme) [inline provider/resource_kpi_threshold_template.go:469]: do we want to keep a warning, or are we ok with being totally silently adding a dummy template?

File: provider/resource_service.go [hunk 3]
```diff
@@ -541,12 +545,30 @@ func (r *resourceService) Update(ctx context.Context, req resource.UpdateRequest
 		return
 	}
 	if existing == nil {
-		resp.Diagnostics.AddError("Unable to update Service", "service not found")
-		return
-	}
-	diags = base.UpdateAsync(ctx)
-	if resp.Diagnostics.Append(diags...); resp.Diagnostics.HasError() {
-		return
+		resp.Diagnostics.AddWarning("Service was not found on the update. Probably it was deleted in the UI.", "Creating...")
```
Comment (benoittoulme) [inline provider/resource_service.go:548]: 👍

File: provider/resource_kpi_threshold_template.go (hunk 2 above)
Comment (masha1997) [reply provider/resource_kpi_threshold_template.go:469]: Thank you, a warning, I added it for service, will update for others as well.

File: provider/resource_service.go (hunk 3 above)
Comment (benoittoulme) [reply provider/resource_service.go:548]: We _could_ just call `func (r *resourceService) Create` here.

File: provider/resource_kpi_threshold_template.go (hunk 2 above)
Comment (masha1997) [reply provider/resource_kpi_threshold_template.go:469]: Updated :)

**Review by benoittoulme — APPROVED** [review]

**Review by dmitrykruglov — APPROVED** [review]

**Review by benoittoulme — APPROVED** [review]

**Pull request merged**